### PR TITLE
Java docs update

### PR DESCRIPTION
**Pull request by maciejwalkowiak**
- bump version to 3.2.0
- update proxy docs (https://github.com/getsentry/sentry-java/pull/1035)
- Update Spring exception resolver order docs (getsentry/sentry-java#1008)
- Update Java configuration (gh-725)

### CODE REVIEW

**Comment by maciejwalkowiak** [comment]
cc @marandaneto @bruno-garcia

File: src/platforms/java/common/configuration/index.mdx [hunk 1]
```diff
@@ -36,6 +36,15 @@ Sentry.init(options -> {
 });
 ```
 
+Whem multiple configuration ways are used, options are resolved in the following order:
```
Comment (marandaneto) [inline src/platforms/java/common/configuration/index.mdx:39]:
```suggestion
When multiple configuration ways are used, options are resolved in the following order:
```

**Review by marandaneto — APPROVED** [review]
looks good to me, thanks for this.

**Comment by pantos27** [comment]
What about performance monitoring docs for Java?

**Comment by maciejwalkowiak** [comment]
@pantos27 this will come next.

**Review by bruno-garcia — APPROVED** [review]
Thank you

**Pull request merged**